lib/glimesh_web/live/streams_live/list.ex

```diff
@@ -24,26 +27,33 @@ defmodule GlimeshWeb.StreamsLive.List do
   end
 
   @impl true
-  def mount(%{"category" => category}, session, socket) do
+  def mount(params, session, socket) do
     if session["locale"], do: Gettext.put_locale(session["locale"])
 
-    case Streams.get_category(category) do
-      %Glimesh.Streams.Category{} = category ->
-        channels = Glimesh.Streams.list_in_category(category)
+    case ChannelCategories.get_category(params["category"]) do
+      %Streams.Category{} = category ->
+        tags = ChannelCategories.list_live_tags(category.id)
 
         {:ok,
          socket
          |> put_page_title(category.name)
          |> assign(:list_name, category.name)
-         |> assign(:channels, channels)
+         |> assign(:tags, tags)
+         |> assign(:tag_selected, Map.has_key?(params, "tag"))
          |> assign(:category, category)}
 
       nil ->
         {:ok, redirect(socket, to: "/")}
     end
   end
 
-  def mount(_, _, socket) do
-    {:ok, redirect(socket, to: "/")}
+  @impl true
+  def handle_params(params, _uri, socket) do
+    channels = ChannelLookups.filter_live_channels(Map.take(params, ["category", "tag"]))
```

Comment:
This is a super neat way of handling page loads, where `mount` isn't called twice. So when you are filtering streams it happens virtually immediately.